crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -176,6 +176,49 @@ def create_proposal
     base
   end
 
+  def prepare_nodes_for_os_upgrade
+    nodes_to_upgrade = []
+    all_nodes = NodeObject.all
+
+    all_nodes.each do |node|
+      next if node.admin? or node[:platform] == "windows"
+      node.set_state("os_upgrading")
+    end
+
+    # wait for the pxe_config to be updated, then reboot the nodes
+    discovery_dir = "#{NodeObject.admin_node[:provisioner][:root]}/discovery/"
+    pxecfg_subdir = "bios/pxelinux.cfg"
+
+    all_nodes.each do |node|
+      next if node.admin? or node["platform"] == "windows"
```

Comment:
Style/AndOr: Use `||` instead of `or`. (https://github.com/bbatsov/ruby-style-guide#no-and-or-or)